File: exchange-ticket/src/main/java/com/web3/exchange/ticket/entity/Ticket.java
package com.web3.exchange.ticket.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.web3.exchange.common.entity.base.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

/**
 * 客服工单（t_ticket）。
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("t_ticket")
public class Ticket extends BaseEntity {
    /** 提交用户ID */
    private Long userId;
    /** 分类：DEPOSIT/WITHDRAW/TRADE/ACCOUNT/OTHER */
    private String category;
    /** 标题 */
    private String title;
    /** 问题描述 */
    private String content;
    /** 状态：0开放 1处理中 2已解决 3已关闭 */
    private Integer status;
    /** 优先级：0低 1中 2高 */
    private Integer priority;
    /** 处理管理员ID */
    private Long assigneeId;
    /** 解决时间 */
    private LocalDateTime resolvedAt;
}

File: frontend/src/api/ticket.ts
import { request } from './http'
import type { PageResult } from './chain'

/** 工单 */
export interface Ticket {
  id?: string | number
  userId?: string | number
  category?: string
  title?: string
  content?: string
  status?: number
  priority?: number
  assigneeId?: string | number
  resolvedAt?: string
  createTime?: string
  updateTime?: string
}

/** 工单回复 */
export interface TicketReply {
  id?: string | number
  ticketId?: string | number
  userId?: string | number
  isStaff?: number
  content?: string
  createTime?: string
}

/** 工单详情 */
export interface TicketDetail extends Ticket {
  replies?: TicketReply[]
}

export const TICKET_CATEGORIES = [
  { label: '充值', value: 'DEPOSIT' },
  { label: '提现', value: 'WITHDRAW' },
  { label: '交易', value: 'TRADE' },
  { label: '账户', value: 'ACCOUNT' },
  { label: '其他', value: 'OTHER' },
]
export const TICKET_STATUS: Record<number, string> = { 0: '待处理', 1: '处理中', 2: '已解决', 3: '已关闭' }

// 用户侧
export function createTicket(data: { category: string; title: string; content: string; priority: number }): Promise<Ticket> {
  return request<Ticket>({ url: '/ticket/create', method: 'post', data })
}
export function myTickets(page = 1, size = 20, status?: number): Promise<PageResult<Ticket>> {
  return request<PageResult<Ticket>>({ url: '/ticket/list', method: 'get', params: { page, size, status } })
}
export function ticketDetail(id: string | number): Promise<TicketDetail> {
  return request<TicketDetail>({ url: `/ticket/${id}`, method: 'get' })
}
export function replyTicket(ticketId: string | number, content: string): Promise<TicketReply> {
  return request<TicketReply>({ url: '/ticket/reply', method: 'post', data: { ticketId, content } })
}
export function closeTicket(id: string | number): Promise<void> {
  return request<void>({ url: `/ticket/${id}/close`, method: 'post' })
}

// 管理侧
export function allTickets(page = 1, size = 20, status?: number): Promise<PageResult<Ticket>> {
  return request<PageResult<Ticket>>({ url: '/admin/ticket/list', method: 'get', params: { page, size, status } })
}
export function adminReplyTicket(ticketId: string | number, content: string): Promise<TicketReply> {
  return request<TicketReply>({ url: '/admin/ticket/reply', method: 'post', data: { ticketId, content } })
}
export function updateTicketStatus(ticketId: string | number, status: number, assigneeId?: string | number): Promise<void> {
  return request<void>({ url: '/admin/ticket/status', method: 'post', data: { ticketId, status, assigneeId } })
}
